From 08273a0d0deae610b93d7f5eb28b0df5f978bf20 Mon Sep 17 00:00:00 2001
Message-Id: <08273a0d0deae610b93d7f5eb28b0df5f978bf20.1569996425.git.liu.denton@gmail.com>
From: Denton Liu <liu.denton@gmail.com>
Date: Tue, 1 Oct 2019 15:51:31 -0700
Subject: [PATCH] t0000: cover GIT_SKIP_TESTS blindspot
To: Git Mailing List <git@vger.kernel.org>

Currently, the tests for GIT_SKIP_TESTS do not cover the situation where
we skip an entire test suite. Add a test case so we cover this
blindspot.

Signed-off-by: Denton Liu <liu.denton@gmail.com>
---
This patch was created as a result of me teaching test-lib.sh to
recognise GIT_RUN_TESTS as a variable, similar to the --run option. That
patch is polished up and ready to go but I realised that running
something like

	make T=t4???-*.sh -j

covers all the use cases I was thinking of so I'm abandoning it. If
anyone has any use for that patch, though, let me know and I can send it
in.

 t/t0000-basic.sh | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/t/t0000-basic.sh b/t/t0000-basic.sh
index 4c01f60dd3..1c5c75920e 100755
--- a/t/t0000-basic.sh
+++ b/t/t0000-basic.sh
@@ -391,6 +391,23 @@ test_expect_success 'GIT_SKIP_TESTS sh pattern' "
 	)
 "
 
+test_expect_success 'GIT_SKIP_TESTS entire suite' "
+	(
+		GIT_SKIP_TESTS='git' && export GIT_SKIP_TESTS &&
+		run_sub_test_lib_test git-skip-tests-entire-suite \
+			'GIT_SKIP_TESTS entire suite' <<-\\EOF &&
+		for i in 1 2 3
+		do
+			test_expect_success \"passing test #\$i\" 'true'
+		done
+		test_done
+		EOF
+		check_sub_test_lib_test git-skip-tests-entire-suite <<-\\EOF
+		> 1..0 # SKIP skip all tests in git
+		EOF
+	)
+"
+
 test_expect_success '--run basic' "
 	run_sub_test_lib_test run-basic \
 		'--run basic' --run='1 3 5' <<-\\EOF &&
-- 
2.23.0.248.g3a9dd8fb08

